src/main/java/playfriends/mc/plugin/features/playerprofile/PlayerProfileHandler.java

```diff
@@ -0,0 +1,97 @@
+package playfriends.mc.plugin.features.playerprofile;
+
+import net.md_5.bungee.api.chat.TextComponent;
+import org.bukkit.configuration.file.FileConfiguration;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.player.AsyncPlayerChatEvent;
+import playfriends.mc.plugin.MessageUtils;
+import playfriends.mc.plugin.api.ConfigAwareListener;
+import playfriends.mc.plugin.playerdata.PlayerData;
+import playfriends.mc.plugin.playerdata.PlayerDataManager;
+
+public class PlayerProfileHandler implements ConfigAwareListener {
+
+    private final PlayerDataManager playerDataManager;
+    private String nameplate;
+    private String message;
+    private String discordSetMessage;
+    private String pronounsSetMessage;
+
+    public PlayerProfileHandler(PlayerDataManager playerDataManager) {
+        System.out.println("PlayerProfileChat created");
```

Comment:
If you want to log something, especially in the async chat method, you should use the plugin's Logger (you can pass it as a constructor argument), otherwise you'll bypass writing to the log file.

Reply:
Will remove since it was for debugging 

(the event wouldn't fire because i was forgetting to recompile the jar after making changes)